server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -689,6 +687,36 @@ public void deleteLogLevel(@PathParam("owner_key") String ownerKey) {
         return page.getPageData();
     }
 
+    private List<ConsumerType> findAndValidateTypeLabels(
```

Comment:
The abstraction to a separate class is a decent first step, but definitely add it to the DI framework if we're going to be using it this way.